Source/Core/Core/IOS/USB/Host.cpp

```diff
@@ -34,7 +36,10 @@ USBHost::~USBHost() = default;
 
 std::optional<IPCReply> USBHost::Open(const OpenRequest& request)
 {
-  if (!m_has_initialised && !Core::WantsDeterminism())
+  auto& system = Core::System::GetInstance();
+  auto& core_timing = system.GetCoreTiming();
+  m_update_event = core_timing.RegisterEvent("UpdateUSBDevices", UpdateDevicesCallback);
```

Comment:
Sorry, I forgot there is a simpler way of running code periodically that doesn't require scheduling CoreTiming events manually. IOS device classes have a function called `Update()` that is called periodically and that can be overridden. So something like this should suffice for our USB device update code:

```cpp
void USBHost::Update()
{
  if (Core::WantsDeterminism())
    UpdateDevices();
}
```

With this new solution you can remove UpdateDevicesCallback() entirely and the `ScheduleEvent` bit you added at the end of UpdateDevices().

(bonus: this doesn't require hardcoding the IOS device path (e.g. `/dev/usb/hid`) and won't crash Dolphin if the user loads a savestate that contains an UpdateUSBDevices event before you have registered it)